src/SFA.DAS.AANHub.Application/Attendances/Commands/CreateAttendance/CreateAttendanceCommandValidator.cs

```diff
@@ -0,0 +1,36 @@
+﻿using FluentValidation;
+using SFA.DAS.AANHub.Application.Common.Validators.RequestedByMemberId;
+using SFA.DAS.AANHub.Domain.Entities;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+
+
+namespace SFA.DAS.AANHub.Application.Attendances.Commands.CreateAttendance;
+public class CreateAttendanceCommandValidator : AbstractValidator<CreateAttendanceCommand>
+{
+    public const string EventIdNotSuppliedMessage = "A Calendar Event Id was not supplied";
+    public const string EventNotFoundMessage = "A calendar event with this ID could not be found";
+    public const string EventInPastMessage = "Cannot attend a calendar event that is in the past";
+    public const string EventCancelledMessage = "Cannot attend a calendar event that is cancelled";
+
+    private CalendarEvent? calendarEvent;
```

Comment:
I'll scope it to the method, since I don't think it needs to be a class-level member.

Calling the repository 3 times was exactly the problem I was trying to avoid here. All 4 paths are covered by tests, but I would like to try and bolster the test coverage a bit more. It's nice to see a specific error message -- one of my pet peeves is when a website tells me "Something bad happened". Besides, the architecture document says that specific errors should be returned for each condition.

Reply:
Now it's scoped to the constructor